apps/challenges/challenge_config_utils.py

```diff
@@ -220,436 +239,757 @@ def get_value_from_field(data, base_location, field_name):
             )
+
+            if isfile(self.challenge_image_path):
+                self.challenge_image_file = ContentFile(
+                    get_file_content(self.challenge_image_path, "rb"), image
+                )
+            else:
+                self.challenge_image_file = None
         else:
-            challenge_image_file = None
-    else:
-        challenge_image_file = None
-    files["challenge_image_file"] = challenge_image_file
+            self.challenge_image_file = None
+        self.files["challenge_image_file"] = self.challenge_image_file
 
-    # Check for challenge description file
-    challenge_config_location = join(
-        BASE_LOCATION, unique_folder_name, extracted_folder_name
-    )
-    is_valid, message = is_challenge_config_yaml_html_field_valid(
-        yaml_file_data, "description", challenge_config_location
-    )
-    if not is_valid:
-        error_messages.append(message)
-    else:
-        yaml_file_data["description"] = get_value_from_field(
-            yaml_file_data, challenge_config_location, "description"
+    def validate_challenge_description(self):
+        self.challenge_config_location = join(
+            self.base_location, self.unique_folder_name, self.extracted_folder_name
         )
+        challenge_description = self.yaml_file_data.get("description")
+        if not challenge_description or len(challenge_description) == 0:
+            message = self.error_messages_dict.get("missing_challenge_description")
+            self.error_messages.append(message)
+        else:
+            is_valid, message = is_challenge_config_yaml_html_field_valid(
+                self.yaml_file_data, "description", self.challenge_config_location
+            )
+            if not is_valid:
+                self.error_messages.append(message)
+            else:
+                self.yaml_file_data["description"] = get_value_from_field(
+                    self.yaml_file_data, self.challenge_config_location, "description"
+                )
 
     # Check for evaluation details file
-    is_valid, message = is_challenge_config_yaml_html_field_valid(
-        yaml_file_data, "evaluation_details", challenge_config_location
-    )
-    if not is_valid:
-        error_messages.append(message)
-    else:
-        yaml_file_data["evaluation_details"] = get_value_from_field(
-            yaml_file_data, challenge_config_location, "evaluation_details"
-        )
+    def validate_evaluation_details_file(self):
+        evaluation_details = self.yaml_file_data.get("evaluation_details")
+        if not evaluation_details or len(evaluation_details) == 0:
+            message = self.error_messages_dict.get("missing_evaluation_details")
+            self.error_messages.append(message)
+        else:
+            is_valid, message = is_challenge_config_yaml_html_field_valid(
+                self.yaml_file_data,
+                "evaluation_details",
+                self.challenge_config_location,
```

Comment:
Same issue for `self.challenge_config_location` as `self.extracted_folder_name`.